src/System.Management.Automation/engine/Subsystem/FeedbackSubsystem/IFeedbackProvider.cs

```diff
@@ -14,6 +13,91 @@
+    {
+        /// <summary>
+        /// Layout for displaying the recommended actions.
+        /// </summary>
+        public enum DisplayLayout
+        {
+            /// <summary>
+            /// Display each recommended action in a row.
+            /// </summary>
+            Portrait,
+
+            /// <summary>
+            /// Display all recommended actions in the same row.
+            /// </summary>
+            Landscape,
+        }
+
+        /// <summary>
+        /// Gets the description message about this feedback.
+        /// </summary>
+        public string Header { get; }
+
+        /// <summary>
+        /// Gets the footer message about this feedback.
+        /// </summary>
+        public string? Footer { get; }
+
+        /// <summary>
+        /// Gets the recommended actions -- command lines or even code snippets to run.
+        /// </summary>
+        public List<string>? RecommendedActions { get; }
+
+        /// <summary>
+        /// Gets the layout to use for displaying the recommended actions.
+        /// </summary>
+        public DisplayLayout Layout { get; }
+
+        /// <summary>
+        /// Gets or sets the next feedback item, if there is one.
+        /// </summary>
+        public FeedbackItem? Next { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FeedbackItem"/> class.
+        /// </summary>
+        /// <param name="header">The description message (must be not null or empty).</param>
+        /// <param name="actions">The recommended actions to take (optional).</param>
+        public FeedbackItem(string header, List<string>? actions)
+            : this(header, actions, footer: null, DisplayLayout.Portrait)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FeedbackItem"/> class.
+        /// </summary>
+        /// <param name="header">The description message (must be not null or empty).</param>
+        /// <param name="actions">The recommended actions to take (optional).</param>
+        /// <param name="layout">The layout for displaying the actions.</param>
+        public FeedbackItem(string header, List<string>? actions, DisplayLayout layout)
+            : this(header, actions, footer: null, layout)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FeedbackItem"/> class.
+        /// </summary>
+        /// <param name="header">The description message (must be not null or empty).</param>
+        /// <param name="actions">The recommended actions to take (optional).</param>
+        /// <param name="footer">The footer message (optional).</param>
+        /// <param name="layout">The layout for displaying the actions.</param>
+        public FeedbackItem(string header, List<string>? actions, string? footer, DisplayLayout layout)
+        {
+            Requires.NotNullOrEmpty(header, nameof(header));
```

Comment:
That's good to know.
I will keep the code as is. It can be changed together if we choose to.

Reply:
It was already done in last month :-) The method is not used anymore.

Reply:
I just took a look. The change was to `ArgumentNullException.ThrowIfNull`. **There was a big problem with that set of changes: the param name is NOT used**. This resulted in all those altered code stopped to have the parameter name information in the exception. Example: https://github.com/PowerShell/PowerShell/pull/18820/files

@iSazonov That's a huge set of changes, and it will take a lot of time to re-do it to add back the parameter names -- are you going to make such changes?
Please be more cautious on changes like this in future -- replace existing code with new APIs, new coding styles, and etc. Those changes are good-to-have, but not essentially needed, and they usually come with changes in many places of the code base, so you need make sure they don't result in any behavior changes.

Reply:
It wasn't an emotional change. I monitor the .Net Runtime repository and their actions to improve code generation. They did this API for two reasons. First, it eliminates errors with the second parameter (nameof). We have fixed this more than once. Also, it is more flexible. For example, Debug.Assert can now automatically include (whole and literal) a conditional expression in the report. Second reason, it makes the code more Runtime friendly to do real-time optimizations.

We could benefit from `CallerArgumentExpression` in code related to `PSArgumentException` etc (and in our custom Assert too). I would not be surprised if we find typos (nameof) there too.

Reply:
@iSazonov Something to improve in future:
- https://github.com/PowerShell/PowerShell/pull/18784
- https://github.com/PowerShell/PowerShell/pull/18792
- https://github.com/PowerShell/PowerShell/pull/18820

None of these 3 PRs mentioned why this change should be made and the benefit for doing this. The PR descriptions should contain more details about this API and the justification for making this change, which would be useful context information for someone who was not involved in the discussion you guys had.

Reply:
Eh, yeah, those descriptions could have been better. Although we can find many times as many worse examples. Nevertheless, I will try to make better descriptions.

Reply:
I have updated those 3 PRs with some context of the API.